language/move-prover/interpreter-testsuite/tests/concrete_check/choice.exp

```diff
@@ -0,0 +1,67 @@
+Running Move unit tests
+[ FAIL    ] 0x2::A::simple_number_min_range_failure
+[ FAIL    ] 0x2::A::simple_number_range_failure
+[ FAIL    ] 0x2::A::vector_choose_min_unsatisfied_predicate
+[ PASS    ] 0x2::A::vector_choose_success
+[ FAIL    ] 0x2::A::vector_choose_unsatisfied_predicate
+
+Test failures:
+
+Failures in 0x2::A:
+
+┌── simple_number_min_range_failure ──────
+│ error: failed to evaluate expression: enumeration of a non-address type domain is not supported
+│    ┌─ tests/concrete_check/choice.move:62:42
+│    │
+│ 62 │         ensures result <= (choose min x: u64 where x >= 4);
+│    │                                          ^^^
+│
+│ error: failed to evaluate expression: unexpected error code
```

Comment:
Is there a way to remove this additional error message?

Reply:
I'm thinking of defining a function similar to `eval_failure_code` to distinguish between domain type failure and other evaluation failures. Then, when we reach the function `check_assert`, we can check the error check to see if it is a domain failure, and stop logging `failed to evaluate expression: unexpected error code` if that is the case?

Reply:
Yes, that is the intended solution. But I just realized that doing this change is outside the scope of this PR (because some of the arithmetic errors are also reported as `unexpected error code`), so let's leave the code as it is now.

Reply:
Yep, I'll make this change in another PR, in that case maybe we can have more informative error message for arithmetic errors too